### DISCUSSION TITLE
How can I check the validity of a given mnemonic phrase using wallet-core

### DISCUSSION THREAD

**Discussion by RobinOnTheWay** [trustwallet/wallet-core#1948] [Q&A]
I tried to check the validity of a given mnemonic phrase by creating a HDWallet object.

``` 
try{
            HDWallet wallet = new HDWallet(mnemonicPhrase,"");
               // mnemonicPhrase is valid if wallet is not null
        }catch (InvalidParameterException e){
             //  mnemonicPhrase is invalid
      }
```
It turned out that even I give it an obviously invalid mnemonic phrase, for example "attack tray total salon drink", there are only 5 words in it, the HDWallet can still be created. No exception has been thrown.

Are there any way of checking the validity of a given mnemonic phrase? Thank you.

**Comment by hewigovens** [comment]
I just tried swift code and it returned `nil`, are you using latest version?
```swift
let words = "attack tray total salon drink"
let wallet = HDWallet(mnemonic: words, passphrase: "")
XCTAssertNil(wallet)
```

**Reply by hewigovens** [reply]
Also tried Kotlin test, please try latest wallet core build 
```kotlin
assertFalse(Mnemonic.isValid("attack tray total salon drink"))
```

**Comment by RobinOnTheWay** [comment]
Hi, @hewigovens , Thank you very much for your reply. 
I checked the interface source once again and found there is a dedicated method in the following to the check the validity of mnemonic phrase and it works well. Thank you.

`public static native boolean isValid(String mnemonic);`

By the way, although you have upgraded wallet-core to [2.6.36](https://github.com/trustwallet/wallet-core/packages/700258?version=2.6.36)
It seems that the latest version on maven repository is still [2.6.3](https://mvnrepository.com/artifact/com.trustwallet/wallet-core) so the latest library can not be imported with maven or gradle. 

I know I can still use the latest version by downloading the library directly from github. It is just a friendly reminder. Thank you.

**Reply by hewigovens** [reply]
Please read https://github.com/trustwallet/wallet-core#android, latest build is hosted on Github Packages

**Comment by RobinOnTheWay** [comment]
@hewigovens 

Thank you for your reply. 
I am now using the latest library from github package. 
As you said, the in the latest version, the following isValid function is gone. And by using one of the HDWallet constructor as in the following, I can check whether the mnemonic phrase is valid. It works well. Thank you.

`public static native boolean isValid(String mnemonic);`

`public HDWallet(String mnemonic, String passphrase, boolean check) `

**Reply by optout21** [reply]
Mnemonic validity check is now in the `Mnemonic.h` file, see:
https://github.com/trustwallet/wallet-core/blob/master/src/Mnemonic.h
